compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -143,6 +143,9 @@ pub enum RuntimePhase {
     /// * [`TerminatorKind::CoroutineDrop`]
     /// * [`Rvalue::Aggregate`] for any `AggregateKind` except `Array`
     /// * [`PlaceElem::OpaqueCast`]
+    /// * [`CastKind::PointerCoercion`] with any of the following:
+    ///   * [`PointerCoercion::ArrayToPointer`]
+    ///   * [`PointerCoercion::MutToConstPointer`]
```

Comment:
compiler/rustc_const_eval/src/interpret/cast.rs should then probably also explicitly panic on these casts.